### PR TITLE
Discrete-continuous sum-mixtures

### PR DESCRIPTION
**Pull request by keevindoherty**
This PR adds a new factor type that computes sum-mixtures of `DCFactors`. Could definitely use a second set of eyes on this.

### CODE REVIEW

File: include/dcsam/DCSAM_utils.h [hunk 1]
```diff
@@ -17,6 +17,23 @@
...
+/*
+ * Numerically stable log-sum-exp function.
+ */
+inline double logSumExp(const std::vector<double> &values) {
+  double max = -std::numeric_limits<double>::infinity();
+  for (size_t i = 0; i < values.size(); i++) {
+    if (values[i] > max) {
+      max = values[i];
+    }
+  }
```
Comment (520xyxyzq) [inline include/dcsam/DCSAM_utils.h:29]: not sure whether using C++ internal functions (e.g. max element, sum, etc.) will improve speed here

Comment (520xyxyzq) [reply include/dcsam/DCSAM_utils.h:29]: https://stackoverflow.com/a/45960829 From a quick google search

**Review by 520xyxyzq — COMMENTED** [review]
The implementation looks good to me. Didn't catch any obvious mistake. But may need more extensive tests to see whether it's doing the right thing.

**Comment by keevindoherty** [comment]
I'm thinking about marking this as stale or moving to an "unstable" dir for the time being, but it looks like there are changes beyond just sum mixtures that we might want to pull into main?

**Comment by keevindoherty** [comment]
Closing for now as stale

**Pull request closed without merging**